src/UI/Implementation/Render/ilJavaScriptBinding.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/* Copyright (c) 2016 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Render;
+
+/**
+ * Wraps global ilTemplate to provide JavaScriptBinding. 
+ */
+class ilJavaScriptBinding implements JavaScriptBinding {
+	/**
+	 * @var \ilTemplate
+	 */
+	private	$global_tpl;
+
+	/**
+	 * @var	integer
+	 */
+	private $id_count = 0;
+
+	public function __construct(\ilTemplate $global_tpl) {
+		$this->global_tpl = $global_tpl;
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function createId() {
+		$this->id_count++;
+		return "id_".$this->id_count;
```

Comment:
please add a more specified prefix here to avoid possible collisions with other id's from other libraries. something like id_il_ui (for id, ilias, ui) or similar should work.